lib/eddystone-beacon-scanner.js

```diff
@@ -24,47 +32,188 @@ EddystoneBeaconScanner.prototype.startScanning = function(allowDuplicates) {
       noble.once('stateChange', startScanningOnPowerOn);
     }
   };
+  
+  var didLoseBeacon = setInterval(this.didLoseBeacon.bind(this), EXIT_GRACE_PERIOD / 2);
 
   startScanningOnPowerOn();
 };
 
 EddystoneBeaconScanner.prototype.stopScanning = function() {
   debug('stopScanning');
+  
   noble.stopScanning();
+  clearInterval(didLoseBeacon);
 };
 
-EddystoneBeaconScanner.prototype.onDiscover = function(peripheral) {
-  debug('onDiscover: %s', peripheral);
+EddystoneBeaconScanner.prototype.didFindBeacon = function(peripheral) {
+  debug('didFindBeacon: %s', peripheral);
+
+  if (this.isEddyStoneBeacon(peripheral) === true) {
 
-  if (this.isUrlBeacon(peripheral)) {
-    var urlBeacon = this.parseUrlBeacon(peripheral);
+    this.cacheBeacon(peripheral);
 
-    debug('onDiscover: urlBeacon = %s', JSON.stringify(urlBeacon));
-    this.emit('discover', urlBeacon);
+    var beaconInfo = this.parseBeacon(peripheral);
+
+    debug('didFindBeacon: Beacon = %s', JSON.stringify(beaconInfo));
+    this.emit('didFindBeacon', beaconInfo);
   }
 };
 
-EddystoneBeaconScanner.prototype.isUrlBeacon = function(peripheral) {
+EddystoneBeaconScanner.prototype.didLoseBeacon = function() {
+  for (var id in inRange) {
+    if (inRange[id].lastSeen < (Date.now() - EXIT_GRACE_PERIOD)) {
+      var peripheral = inRange[id].peripheral;
+      var lostBeacon = this.parseBeacon(peripheral);
+
+      this.emit('didLoseBeacon', lostBeacon);
+
+      delete inRange[id];
+    }
+  }
+};
+
+EddystoneBeaconScanner.prototype.isEddyStoneBeacon = function(peripheral) {
   var serviceData = peripheral.advertisement.serviceData;
 
   // make sure service data is present, with the expected uuid and data length
   return ( serviceData &&
            serviceData.length > 0 &&
            serviceData[0].uuid === SERVICE_UUID &&
-           serviceData[0].data.length > 2 &&
-           serviceData[0].data[0] === 0x10
+           serviceData[0].data.length > 2
          );
 };
 
-EddystoneBeaconScanner.prototype.parseUrlBeacon = function(peripheral) {
+EddystoneBeaconScanner.prototype.cacheBeacon = function(peripheral) {
+  var id = peripheral.id;
+  var entered = !inRange[id];
```

Comment:
Do you really only want to cache it the first time you see it? not the most recent one you've seen? Maybe just cache the peripheral.id for identification? What is the use case for didlose besides knowing which one disappeared?